### ISSUE TITLE
Add ECDSA signing functionality

### ISSUE THREAD

**Issue by indomitableSwan** [boltlabs-inc/key-mgmt-spec#107]
We need to be able to create and use ECDSA keys. 

This involves:
- [x] The option to create a specific key type during generate, not just arbitrary secrets.
- [x] Protocol(s) to request signatures on a submitted message.
    - [x] Remotely (higher priority)
    - [x] Locally (lower priority, OK to document as an issue and leave in backlog).

Whoever closes this ticket should create the appropriate issues for implementation in key-mgmt.

**Comment by indomitableSwan** [comment]
This ticket can be closed once PR #119 is merged and relevant tickets have been created in key-mgmt.

**Comment by indomitableSwan** [comment]
Closing as superseded by https://github.com/boltlabs-inc/key-mgmt/issues/203.
